airbyte-integrations/connectors/source-sendgrid/source_sendgrid/manifest.yaml

```diff
@@ -1,257 +1,981 @@
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - result
       paginator:
-        $ref: "#/definitions/offset_paginator"
-  - $ref: "#/definitions/base_stream"
-    $parameters:
-      name: "messages"
-      primary_key: "msg_id"
-      cursor_field: "last_event_time"
-      path: "/v3/messages"
-      field_path: []
-    incremental_sync:
-      $ref: "#/definitions/incremental_sync"
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestPath
+        page_size_option:
+          type: RequestOption
+          field_name: page_size
+          inject_into: request_parameter
+        pagination_strategy:
+          type: CursorPagination
+          page_size: 100
+          cursor_value: '{{ response.get("_metadata", {}).get("next", {}) }}'
+          stop_condition: '{{ not response.get("_metadata", {}).get("next", {}) }}'
+      partition_router: []
+  - type: DeclarativeStream
+    name: templates
+    primary_key: []
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $schema: http://json-schema.org/schema#
+        properties:
+          generation:
+            type: string
+          id:
+            type: string
+          name:
+            type: string
+          updated_at:
+            format: date-time
+            type: string
+          versions:
+            type: array
+        type: object
     retriever:
-      $ref: "#/definitions/base_stream/retriever"
+      type: SimpleRetriever
       requester:
-        $ref: "#/definitions/requester"
+        type: HttpRequester
+        url_base: https://api.sendgrid.com
+        path: /v3/templates
+        http_method: GET
         request_parameters:
-          limit: "1000"
-          query: 'last_event_time BETWEEN TIMESTAMP "{{stream_slice.start_time}}" AND TIMESTAMP "{{stream_slice.end_time}}"'
-  - $ref: "#/definitions/base_stream"
-    $parameters:
-      name: "unsubscribe_groups"
-      primary_key: "id"
-      path: "/v3/asm/groups"
-      field_path: []
-check:
-  type: CheckStream
-  stream_names: ["lists"]
+          generations: legacy,dynamic
+        request_headers: {}
+        authenticator:
+          type: BearerAuthenticator
+          api_token: "{{ config['api_key'] }}"
+        request_body_json: {}
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - result
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestPath
+        page_size_option:
+          type: RequestOption
+          field_name: page_size
+          inject_into: request_parameter
+        pagination_strategy:
+          type: CursorPagination
+          page_size: 200
+          cursor_value: '{{ response.get("_metadata", {}).get("next", {}) }}'
+          stop_condition: '{{ not response.get("_metadata", {}).get("next", {}) }}'
+      partition_router: []
```

Comment:
Why do we need an empty `partition_router` here?

Reply:
This is how the builder does it. Seems fine

Reply:
Removed all empty values, please take a look